File: mario.c
#include <stdio.h>
#include <cs50.h>

 
int pyramid(int height);

int main(void)
{
 int height;
 
    // prompts user input
    do
    {
        printf("Height:");
        height = get_int();
    }
    while ((height < 0) || (height > 23));
 
    //calls pyramid function
    pyramid(height);
}

    // loop user input to create rows, spaces, and hashes
int pyramid(int height)
{
    int rows;
    int space;
    int hash;
    for (rows = 1; rows <= height; rows++) 
    {
        for (space = (height - rows); space > 0; space--)
        {
            printf(" "); 
        }
 
        for (hash = 0; hash <= rows; hash++)
        {   
            printf("#"); 
        }
 
        printf("\n");
    }
    return 0;
}

File: mario.py
import cs50

def main():
    """Prompt user for input and call Pyramid."""

    # prompts user input
    while True:
        print("Height:", end ="")
        height = cs50.get_int()
        if height > 0 and height <= 23:
            break
    # calls pyramid method
    pyramid(height)


def pyramid(height):
    """Draw Marios pyramid."""
    
    # draw pyramid increasing #, and decreasing spaces by height
    for i in range(1,height + 1,1):
        for n in range(height - i, 0, -1):
            print(" ", end="")
        for l in range(0, i + 1, 1):
            print("#", end="")
        print("")
    # success
    exit(0)


if __name__ == "__main__":
    main()
